airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_options/request_options_provider.py

```diff
@@ -28,7 +28,7 @@ def get_request_params(
         stream_state: Optional[StreamState] = None,
         stream_slice: Optional[StreamSlice] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
-    ) -> MutableMapping[str, Any]:
+    ) -> Mapping[str, Any]:
```

Comment:
This in theory could have an affect if we were relying on manipulating the contents of this return value. Do you by chance have context why this specific return was originally mutable?